### PR TITLE
contacts bank - added search bar

### PR DESCRIPTION
**Pull request by y-golde**
added a functional search bar to contacts bank

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankForm.tsx [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+import ContactsBankSearchBar from './ContactsBankSearchBar/ContactsBankSearchBar';
 
 const headline = 'בנק מגעים:';
 
 const ContactsBankForm = (props: Props) => {
     const { existingPersons } = props;
     const classes = useStyles(); 
     const {contactBank} = useContext(contactBankContext);
-    
+    const [query, setQuery] = useState<string>("");
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankForm.tsx:17]: change to single quotes

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  onSearchClick: (searchQuery : string) => void; 
+}
+
+const stringAlphabet = yup
+  .string()
+  .matches(ALPHANUMERIC_SLASHES_TEXT_REGEX, errorMessage)
+  .max(50, maxLengthErrorMessage);
+
+const TableSearchBar = (props: Props) => {
+    const [searchQuery, setsearchQueries] = useState<string>("");
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx:26]: same

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        <TypePreventiveTextField
+            {...rest}
+            name='tableSearchBar'
+            className={classes.searchBar}
+            value={searchQuery}
+            validationSchema={stringAlphabet}
+            InputProps={{
+            endAdornment: (
+                <InputAdornment position='end'>
+                    <IconButton id="searchIconButton" onClick={() => onSearchClick(searchQuery)}>
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx:41]: same

**Review by Noam6421 — DISMISSED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankForm.tsx (hunk 1 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankForm.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1363/commits/84f1f190092001b223eb1e4c1b6419d0f91894c5)

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx (hunk 2 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1363/commits/84f1f190092001b223eb1e4c1b6419d0f91894c5)

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx (hunk 3 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/ContactsBank/ContactsBankSearchBar/ContactsBankSearchBar.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1363/commits/84f1f190092001b223eb1e4c1b6419d0f91894c5)

**Review by chenFriedman — APPROVED** [review]

**Review by Noam6421 — APPROVED** [review]

**Pull request merged**